sentence_transformers/sparse_encoder/SparseEncoder.py

```diff
@@ -636,36 +648,10 @@ def get_sentence_embedding_dimension(self) -> int | None:
 
     @contextmanager
     def truncate_sentence_embeddings(self, truncate_dim: int | None) -> Iterator[None]:
-        """
-        In this context, :meth:`SparseEncoder.encode <sentence_transformers.sparse_encoder.SparseEncoder.encode>` outputs
-        sentence embeddings with at most ``truncate_dim`` number of active dimensions.
-
-        This may be useful when you are using a model but want to have a treshold for the number of active dimensions
-        in the output to control the size of the embeddings you want to index.
-
-        Args:
-            truncate_dim (int, optional): The dimension to truncate sentence embeddings to. ``None`` does no truncation.
-
-        Example:
-            ::
-
-                from sentence_transformers import SparseEncoder
-
-                model = SparseEncoder("naver/splade-cocondenser-ensembledistil")
-
-                with model.truncate_sentence_embeddings(truncate_dim=16):
-                    embeddings_truncated = model.encode(["hello there", "hiya"])
-
-                for i in range(embeddings_truncated.shape[0]):
-                    num_nonzero = embeddings_truncated[i].coalesce().indices().shape[1]
-                    assert num_nonzero <= 16, f"Embedding {i} has {num_nonzero} non-zero elements"
-        """
-        original_output_dim = self.truncate_dim
-        try:
-            self.truncate_dim = truncate_dim
-            yield
-        finally:
-            self.truncate_dim = original_output_dim
+        raise NotImplementedError(
+            "SparseEncoder does not support truncating sentence embeddings. "
+            "Use the `max_active_dims` parameter in the encode method instead."
```

Comment:
```suggestion
            "Use the `max_active_dims` parameter in the encode method instead if you want to limit the embedding memory usage."
```
Maybe something like this to be a bit more explicit?